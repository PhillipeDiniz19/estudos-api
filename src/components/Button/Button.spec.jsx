import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./Button";
import userEvent from "@testing-library/user-event";

describe("<Button/>", () => {
  it("should render the button with the text", () => {
    // dentro da aspas é descrição do meu teste
    render(<Button text="Load more" />); // aqui eu renderizo meu componente para teste

    expect.assertions(1); // Nesse caso eu falo que vai ter certa quantidade de erros para o teste.

    const button = screen.getByRole("button", { name: /load more/i }); // Estou criando uma const para receber o meu elemento na tela junto com o texto

    // expect(button).toHaveAttribute('class', 'button') // Aqui meu teste ver se no meu button eu tenho a class com o nome.
    expect(button).toBeInTheDocument(); // Espero que meu elemento estaja na tela se não teste falha.
  });

  it("should call function on button click", () => {
    render(<Button text="Load more" />);
    const fn = jest.fn(); // cria um mocker meio que substituir função existente com uma versão simulada
    const button = screen.getByRole("button", { name: /load more/i });

    userEvent.click(button); // faz a mesma coisa de baixo
    // fireEvent.click(button)  aqui eu crio um evento para disparar um click no button

    expect(fn).toHaveBeenCalledTimes(0);
  });

  it("should be disable when disable is true", () => {
    render(<Button text="Load more" disabled={true} />);

    const button = screen.getByRole("button", { name: /load more/i });

    expect(button).toBeDisabled();
  });

  it("should be disable when disable is false", () => {
    render(<Button text="Load more" disabled={false} />);

    const button = screen.getByRole("button", { name: /load more/i });

    expect(button).toBeEnabled();
  });

  it("should button snpshot teste", () => {
    const fn = jest.fn();
    const { container } = render(<Button text="Load more" disabled={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
