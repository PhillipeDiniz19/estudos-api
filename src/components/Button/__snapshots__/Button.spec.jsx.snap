// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should button snpshot teste 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
